crates/containerd-shim-wasm/src/sandbox/instance.rs

```diff
@@ -107,7 +104,9 @@ impl<Engine: Send + Sync + Clone> InstanceConfig<Engine> {
 
 /// Represents a WASI module(s).
 /// Instance is a trait that gets implemented by consumers of this library.
-pub trait Instance {
+/// This trait requires that any type implementing it is `'static`, similar to `std::any::Any`.
+/// This means that the type cannot contain a non-`'static` reference.
+pub trait Instance: 'static {
```

Comment:
This is so that we can move an `Arc<Instance>` into a thread.
We call `instance.wait()` in a thread since it's now a blocking call.

The `'static` here doesn't mean that any `Instance` variable has to be a global static variable. It does mean that you *could* if you wanted, create a global static of that type.
For example, a `String` would satisfy `'static`, as would `'static str`. However `'a str` would not.
Anything we think of as an _owned_ type, would satisfy `'static`, and `'static` references too. But non `'static` references would not.

The alternative to this is having a _shared state_ object that the `Instance` can return to signal the end of the process (similar-ish to what we had before). What I mean by _shared state_ is for example, and `mpsc::channel` or even a clone of the `WaitableCell`. Before this change the `Mutex` + `CondVar` tuple was the shared state.

I personally prefer imposing `'static` (which I think is a low price to pay) and having a simpler API, but I'd love to hear your thoughts.